lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,40 @@
+export abstract class GoogleError extends Error {
```

Comment:
nit/non-blocking: I might vote to try and avoid subclassing / abstract classes as I feel like it adds a good amount of boilerplate / complextity, since we now need to ensure constructor params are passed down through the child chain.

My preference would be to express things in a more functional way like this using something like a [Discriminated Union type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions). For example,

```typescript
enum GoogleErrorType {
    InvalidGrant,
    UnauthorizedClient
}

type GoogleError
  = InvalidGrant
  | UnauthorizedClient;


interface InvalidGrant {
    kind: GoogleErrorType.InvalidGrant;
    err: Error;
}

interface UnauthorizedClient {
    kind: GoogleErrorType.UnauthorizedClient;
    err: Error;
}


/**
  *  creating errors
  */
function createError<T extends GoogleErrorType>(
  kind: T,
  err: Error = new Error()
) {
  return { kind, err };
}

/**
 * example desriminating
 */
function handleGoogleError(googErr: GoogleError) {
    switch (googErr.kind) {
        case GoogleErrorType.InvalidGrant: {
            /**
             * here the type system knows that
             * the error is an invalid grant,
             * as this compiles
             */
            let e: InvalidGrant = googErr;
        }
    }
}
```